src/client/client.cpp

```diff
@@ -1787,18 +1787,89 @@ void Client::makeScreenshot()
 	if (fs::IsPathAbsolute(g_settings->get("screenshot_path")))
 		screenshot_dir = g_settings->get("screenshot_path");
 	else
-		screenshot_dir = porting::path_user + DIR_DELIM + g_settings->get("screenshot_path");
+		screenshot_dir = porting::path_user + std::string(DIR_DELIM) + g_settings->get("screenshot_path");
+
+	// Using Irrlicht methods since those defined in filesys.cpp are
```

Comment:
This is gross, and should probably be moved to filesys.cpp or at least another function

Reply:
I'm expecting at least two functions here: `getAbsolutePath(path)` and `joinPath(base, filename)`

Reply:
No longer using irrlicht. Going to test already defined function in filesys.cpp and see if it works instead. If it does not i'll try to fix it.

Reply:
Also the unix function realpath (used by a function in filesys.cpp) does not return correct path when source path included non-existing files and/or directories, I'm trying to make a workaround for it.

Reply:
Irrlicht has a compatible license, so you could look to see how it implemented it

Reply:
Implemented a working workaround in filesys.cpp, also fixed a bug in RemoveRelativePathComponents where it would not work correctly with path that is drive root, absolute path should now be generated correctly.